hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsRestOperation.java

```diff
@@ -185,13 +189,18 @@ void execute() throws AzureBlobFileSystemException {
       try {
         LOG.debug("Retrying REST operation {}. RetryCount = {}",
             operationType, retryCount);
+
         Thread.sleep(client.getRetryPolicy().getRetryInterval(retryCount));
       } catch (InterruptedException ex) {
         Thread.currentThread().interrupt();
       }
     }
 
     if (result.getStatusCode() >= HttpURLConnection.HTTP_BAD_REQUEST) {
+      if (this.isAppendBlobAppend && retryCount > 0 && result.getStorageErrorCode().equals("InvalidQueryParameterValue")) {
```

Comment:
Why is it that for Http Status code 400 and above, exception is suppressed ? Can you please add code comments for the reason.

Reply:
This isnt the right place to handle a case specific to appendblob. HttpOperation returned to specific AbfsClient method can take the call on what to return. Similar is done for rename.
You can check: 
https://github.com/apache/hadoop/blob/trunk/hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsClient.java#L358